src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,448 @@
+                    const Utils::Registry packageReg(m_key, m_userId + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp);
+                    m_version = fields.at(INDEX_VERSION);
+                    m_architecture = getArchitecture(fields.at(INDEX_ARCHITECTURE));
+                    m_location = getLocation(packageReg);
+                    m_installTime = getInstallTime(fields);
+                    m_name = getName(fields.at(INDEX_NAME), packageReg);
+                    m_vendor = getVendor(packageReg);
+
+                }
+
+                /*
+                 * If the package location is not in application store database provided by Windows then it is thrown away.
+                 * https://answers.microsoft.com/en-us/windows/forum/all/what-is-cprogram-fileswindows-apps-hidden-folder/783b5a18-c44d-46f7-b638-e98054b7c2a8
+                 */
+                if (m_location.find(STORE_APPLICATION_DATABASE) == std::string::npos)
+                {
+                    m_name.clear();
+                    m_vendor.clear();
+                    m_architecture.clear();
+                    m_version.clear();
+                    m_location.clear();
+                }
+            }
+        }
+
+        /*
+         * @brief Gets packages architecture.
+         *
+         * @param[in] field Field where the architecture type is.
+         *
+         * Type are:
+         *          - x64
+         *          - x86
+         *
+         * @return Return app architecture.
+         */
+        const std::string getArchitecture(const std::string& field)
+        {
+            std::string architecture { UNKNOWN_VALUE };
+
+            if (!field.compare("x64"))
+            {
+                architecture = "x86_64";
+            }
+            else if (!field.compare("x86"))
+            {
+                architecture = "i686";
+            }
+
+            return architecture;
+        }
+
+        /*
+         * Getting location where is the data and configuration application.
+         *
+         * @param[in] registry  Registry object pointing to where the data is.
+         *
+         * @return Return package location.
+         */
+        const std::string getLocation(const Utils::Registry& registry)
+        {
+            std::string location;
+
+            return registry.string("PackageRootFolder", location) ? location : UNKNOWN_VALUE;
+        }
+
+
+        const std::string getInstallTime(const std::vector<std::string>& partName)
+        {
+            constexpr auto WINDOWS_UNIX_EPOCH_DIFF_SECONDS { 11644473600ULL };
+            std::string installTime { DEFAULT_VALUE_STRING };
+            unsigned long long int value;
+            /*
+             * partName varialble has the following fields in order
+             *
+             * vector<string> = {app_Name, app_version, app_Architecture, app_UUID}
+             *
+             * Name of main registry is: app_Name + _ + app_UUID
+             */
```

Comment:
This comment is a red flag about partName variable self-description, used in this method and other ones.
Consider creating a object/struct with named attributes rather than implicit-meaning and error-prone vectors.